### ISSUE TITLE
Update from Open Layers 6 to 8 causes AnimatedCluster animation glitch

### ISSUE THREAD

**Issue by nerdess** [Viglino/ol-ext#1011]
I recently updated from Open Layers 6 to Open Layers 8.

I am using an AnimatedCluster() and since the update, the transition is not nice anymore. The icons are bloated up during transitioning....

Before (all good): 

https://github.com/Viglino/ol-ext/assets/711925/190acb70-c7a5-4640-a3b1-66597c9c9a91

Now (icon scale broken while transitioning):

https://github.com/Viglino/ol-ext/assets/711925/8c9611c5-5898-4ac4-9479-de4948e895a5

Any hints what could cause this would be great :)

**Comment by Viglino** [comment]
It's seems there is a scale factor on the icons...
Do you have an online example?

**Comment by mike-000** [comment]
Possibly due to repeatedly reloading images (which is asynchronous) if the total number of icon sources exceeds the built in cache.  You could define your own style caches outside the style function
```
const circleCache = {};
const iconCache = {};
```
Then in the style function simply look up the color or src, only creating a new style and adding to the cache if it does not exist, then you only need set the scale used to render it
```
const color = hover || selected ? '#1F54A0' : 'rgba(51, 51, 51, 0.8)';
let circle = circleCache[color];
if (!circle) {
	circle = new olStyle.Circle({
		radius: 30 * 1.5;,
		fill: new olStyle.Fill({
			color: color,
		}),
	});
	circleCache[color] = circle;
}

const src = ICONS[category_tag || 'uncategorized'].src;
let icon = iconCache[src];
if (!icon) {
	icon = new olStyle.Style({
		image: new olStyle.Icon({
			src: src,
		}),
	}),
	iconCache[src] = icon;
}

const scale = Math.min(0.4, 1 / resolution);
circle.getImage().setScale(scale);
icon.getImage().setScale(scale);

return [circle, icon];
```

**Comment by Viglino** [comment]
May be related to #1013

**Comment by nerdess** [comment]
The fix introduced here #1013 was also fixing my issue. Hope there will be a release soon to add it to the official codebase :)
